tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/ModelChecker.scala

```diff
@@ -506,22 +510,38 @@ class ModelChecker(typeFinder: TypeFinder[CellT],
     }
   }
 
-  private def dumpCounterexample(): String = {
-    val filename = "counterexample.txt"
+  private def dumpCounterexample(notInv: TlaEx): String = {
+    // Assumption: stack cannot be empty
+    val filename = if(tlc) "MC.out" else "counterexample.tla"
```

Comment:
Let's delegate the name choice to a specific implementation. Note that in the multicore version, we have to give a stem, as multiple core may produce multiple counterexamples for multiple invariants.